src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -39,12 +40,25 @@ def create(self, username, password):
         user_data = AuthManager.authenticate(username, password)
         user_perms, pwd_expiration_date, pwd_update_required = None, None, None
         max_attempt = Settings.ACCOUNT_LOCKOUT_ATTEMPTS
+        mgr_map = mgr.get('mgr_map')
+        url = mgr_map.get('services', {}).get('dashboard', None)
         if max_attempt == 0 or mgr.ACCESS_CTRL_DB.get_attempt(username) < max_attempt:
             if user_data:
                 user_perms = user_data.get('permissions')
                 pwd_expiration_date = user_data.get('pwdExpirationDate', None)
                 pwd_update_required = user_data.get('pwdUpdateRequired', False)
 
+            if isinstance(Settings.MULTICLUSTER_CONFIG, str):
+                item_to_dict = json.loads(Settings.MULTICLUSTER_CONFIG)
+                copy_config = item_to_dict.copy()
+            else:
+                copy_config = Settings.MULTICLUSTER_CONFIG.copy()
+            try:
+                copy_config['config'].append({'name': mgr.get('config')['fsid'], 'url': url, 'cluster_alias': 'local-cluster'})
+            except KeyError:
+                copy_config = {'current_url': url, 'config': [{'name': mgr.get('config')['fsid'], 'url': url, 'cluster_alias': 'local-cluster'}]}
```

Comment:
if you just add a check like
```
if 'config' not in multicluster_config:
    multicluster_config['config'] = {}
```
then you don't need to duplicate the copy_config assigning, which feels like not a clean code.